### PR TITLE
Scan jars for possible Android dependencies

### LINKED ISSUE #780: Aar artifacts with dependencies
If an aar was built with external dependencies (like BarcodeScan for Android), these dependencies have to be added to the android_project [build](https://github.com/gluonhq/substrate/blob/master/src/main/resources/native/android/android_project/app/build.gradle#L11):

```
dependencies {
        // TODO: include AAR dependencies
        implementation 'com.android.support:support-v4:28.0.0'
        api fileTree(dir: '../libs', include: '*.aar')
        api fileTree(dir: '../libs', include: '*.jar')
    }
```

### PR DESCRIPTION
**Pull request by jperedadnr**
Fixes #780 

This PR adds a way to inject dependencies in the Android_Project/app/build.gradle file, by scanning jars in the classpath.

This allows Attach and other third party libraries to include transitive dependencies and not only the classes in the jar/aar files.

### CODE REVIEW

File: src/main/java/com/gluonhq/substrate/config/ConfigResolver.java [hunk 1]
```diff
@@ -194,6 +195,21 @@ public ConfigResolver(String classpath) throws IOException, InterruptedException
...
     }
 
+    /**
+     * Walks through the jars in the classpath,
+     * and looks for META-INF/substrate/config/android-dependencies.txt file.
+     *
+     * @return a list of dependencies lines that should be added to the build.gradle file
+     * @throws IOException Exception while reading the file.
+     */
+    public Set<String> getAndroidDependencies() throws IOException {
```
Comment (johanvos) [inline src/main/java/com/gluonhq/substrate/config/ConfigResolver.java:205]: This should probably go to an Android specific class

File: src/main/java/com/gluonhq/substrate/config/ConfigResolver.java [hunk 2]
```diff
@@ -194,6 +195,21 @@ public ConfigResolver(String classpath) throws IOException, InterruptedException
...
+        Logger.logDebug("Scanning for android dependencies files");
+        return Set.copyOf(scanJars(USER_ANDROID_DEPENDENCIES_FILE,
+                null,
```
Comment (johanvos) [inline src/main/java/com/gluonhq/substrate/config/ConfigResolver.java:208]: Probably better to add a 2-args scanJars method instead of passing null

**Comment by eugener** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/gluonhq/substrate/config/AndroidResolver.java  6
         

Clones added
============
- src/main/java/com/gluonhq/substrate/config/AndroidResolver.java  1
- src/main/java/com/gluonhq/substrate/config/ConfigResolver.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=6276421&bid=20429255)

**Review by tiainen — APPROVED** [review]

**Comment by eugener** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/gluonhq/substrate/config/AndroidResolver.java  6
         

Clones added
============
- src/main/java/com/gluonhq/substrate/config/AndroidResolver.java  1
- src/main/java/com/gluonhq/substrate/config/ConfigResolver.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/gluonhq/substrate/pullRequest?prid=6276421&bid=20429255)

**Review by johanvos — APPROVED** [review]

**Pull request merged**